src/software/thunderscope/thunderscope.py

```diff
@@ -247,34 +498,168 @@ def close(self):
+        default="/tmp/tbots",
+    )
+    parser.add_argument(
+        "--blue_fullsystem_runtime_dir",
+        type=str,
+        help="blue fullsystem runtime directory",
+        default="/tmp/tbots/blue",
+    )
+    parser.add_argument(
+        "--yellow_fullsystem_runtime_dir",
+        type=str,
+        help="yellow fullsystem runtime directory",
+        default="/tmp/tbots/yellow",
+    )
 
-        log_dock = thunderscope.setup_log_widget()
-        thunderscope.dock_area.addDock(log_dock)
+    # Debugging
+    parser.add_argument(
+        "--debug_fullsystem",
+        action="store_true",
+        default=False,
+        help="Debug fullsystem",
+    )
+    parser.add_argument(
+        "--debug_simulator",
+        action="store_true",
+        default=False,
+        help="Debug the simulator",
+    )
 
-        drive_and_dribbler_dock = thunderscope.setup_drive_and_dribbler_widget()
-        thunderscope.dock_area.addDock(drive_and_dribbler_dock)
-        thunderscope.show()
+    # Run blue or yellow full system over WiFi
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--run_blue",
+        action="store_true",
+        help="Run full system as the blue team, over WiFi; estop required",
+    )
+    group.add_argument(
+        "--run_yellow",
+        action="store_true",
+        help="Run full system as the yellow team, over WiFi; estop required",
+    )
+    parser.add_argument(
+        "--interface",
+        action="store",
+        type=str,
+        default=None,
+        help="Which interface to communicate over",
+    )
 
-    elif args.run_simulator:
-        print(
-            "TODO #2050, this isn't implemented, just run the current standalone simulator"
-        )
+    # Sanity check that an interface was provided
+    args = parser.parse_args()
 
+    if args.run_blue or args.run_yellow:
+        if args.interface is None:
+            parser.error("Must specify interface")
+
+    tscope = Thunderscope()
+
+    ###########################################################################
+    #              AI + Robot Communication + Robot Diagnostics               #
+    ###########################################################################
+    #
+    # When we are running with real robots. We want to run 1 instance of AI
+    # and 1 instance of RobotCommunication which will send/recv packets over
+    # the provided multicast channel.
+    proto_unix_io = tscope.blue_full_system_proto_unix_io
+    runtime_dir = args.blue_fullsystem_runtime_dir
+    friendly_colour_yellow = False
+
+    if args.run_yellow:
+        proto_unix_io = tscope.yellow_full_system_proto_unix_io
+        runtime_dir = args.yellow_fullsystem_runtime_dir
+        friendly_colour_yellow = True
+
+    if args.run_blue or args.run_yellow:
+        # TODO (#2585): Support multiple channels
+        with RobotCommunication(
+            proto_unix_io, ROBOT_MULTICAST_CHANNEL_0, args.interface
```

Comment:
why is it `ROBOT_MULTICAST_CHANNEL_0` instead of a list?

Reply:
see #2585, its annoying to change now because of the char[][] array, in the near future we will nuke firmware and support more than 1 channel because we can use a map and bind that to a dictionary